src/types.zig

```diff
@@ -279,23 +292,55 @@ pub const Function = struct {
 
         _ = try writer.writeByteNTimes(' ', indent + 4);
 
-        var split = std.mem.splitSequence(u8, namespace.items, "_");
-        try writer.writeAll(split.first());
-        while (split.next()) |name| {
-            try writer.writeAll(".");
-            try writer.writeAll(name);
+        const is_ret_obj = fun.return_ty.info == .composite_ref;
+
+        const is_const = if (fun.parent) |parent|
+            if (parent.name) |name|
+                std.mem.eql(u8, name, fun.return_ty.slice)
+            else
+                false
+        else
+            false;
+
+        const is_method = if (fun.params.len > 0) if (fun.parent) |parent|
+            if (parent.name) |name|
+                std.mem.eql(u8, name, fun.params[0].type.slice)
+            else
+                false
+        else
+            false else false;
```

Comment:
Feels like we should be able to write these blocks with less `if` - hard to read. Maybe a `:blk`?